src/main/java/uk/gov/pay/products/service/PaymentDeleter.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.pay.products.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.products.config.ExpungeHistoricalDataConfig;
+import uk.gov.pay.products.persistence.dao.PaymentDao;
+
+import javax.inject.Inject;
+import java.time.Clock;
+import java.time.temporal.ChronoUnit;
+
+import static java.lang.String.format;
+import static java.time.ZoneOffset.UTC;
+
+public class PaymentDeleter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PaymentDeleter.class);
+    
+    private ExpungeHistoricalDataConfig expungeHistoricalDataConfig;
+    private PaymentDao paymentDao;
+    private Clock clock;
+
+    @Inject
+    public PaymentDeleter(ExpungeHistoricalDataConfig expungeHistoricalDataConfig, PaymentDao paymentDao, Clock clock) {
+        this.expungeHistoricalDataConfig = expungeHistoricalDataConfig;
+        this.paymentDao = paymentDao;
+        this.clock = clock;
+    }
+
+    public void deletePayments() {
+        if (!expungeHistoricalDataConfig.isExpungeHistoricalDataEnabled()) {
+            LOGGER.info("Expunging of historical data is not enabled.");
+            return;
+        }
+        
+        var maxDate = clock.instant().minus(expungeHistoricalDataConfig.getExpungeDataOlderThanDays(), ChronoUnit.DAYS).atZone(UTC);
+        int numberOfDeletedPayments = paymentDao.deletePayments(maxDate, expungeHistoricalDataConfig.getNumberOfTransactionsToExpunge());
+        LOGGER.info(format("%s payments were deleted.", numberOfDeletedPayments));
```

Comment:
Can be in a separate PR. Can `no_of_payments_deleted` be logged as key-value pair so we can query easily in splunk

Reply:
I'll do this in the next PR.